lib/collection/src/collection/mod.rs

```diff
@@ -454,13 +454,15 @@ impl Collection {
             Some(ReplicaState::Resharding | ReplicaState::ReshardingScaleDown)
         );
 
+        // Marks replica as dead
+        replica_set
+            .ensure_replica_with_state(peer_id, new_state)
+            .await?;
+
         if is_resharding && new_state == ReplicaState::Dead {
             drop(shard_holder);
 
-            let resharding_state = self
-                .resharding_state()
-                .await
-                .filter(|state| state.peer_id == peer_id);
+            let resharding_state = self.resharding_state().await;
```

Comment:
I suggest to add this as minor remark, the plan is to eventually support multiple concurrent operations.

```suggestion
            // Pick the resharding state that is related to this replica
            // Currently we can have one operation at a time thus we pick the main one
            let resharding_state = self.resharding_state().await;
```

Reply:
I'd leave it out for now, cause this is more confusing than helpful (imo) and we might *not* add concurrent resharding support for the next 2 years for all we know, and it will just stay there. 😂